module-apps/apps-common/ApplicationCommonPopupBlueprints.cpp

```diff
@@ -0,0 +1,116 @@
+#include "service-db/Settings.hpp"
+#include "service-db/agents/settings/SystemSettings.hpp"
+#include "popups/data/PopupData.hpp"
+
+namespace app
+{
+    void ApplicationCommon::registerPopupBlueprints()
+    {
+        using namespace gui::popup;
+        popupBlueprint.registerBlueprint(
+            ID::PhoneModes, [&](gui::popup::ID /*id*/, std::unique_ptr<gui::PopupRequestParams> &params) {
+                auto popupParams = dynamic_cast<gui::PhoneModePopupRequestParams *>(params.get());
+                if (popupParams == nullptr) {
+                    return false;
+                }
+                const auto &mode = popupParams->getPhoneMode();
+                auto flightModeSetting =
+                    settings->getValue(settings::Cellular::offlineMode, settings::SettingsScope::Global);
+                bool flightMode = flightModeSetting == "1";
+
+                const auto &popupName = resolveWindowName(gui::popup::ID::PhoneModes);
+                if (const auto currentWindowName = getCurrentWindow()->getName(); currentWindowName == popupName) {
+                    updateCurrentWindow(std::make_unique<gui::ModesPopupData>(mode, flightMode));
+                }
+                else {
+                    switchWindowPopup(popupName,
+                                      popupParams->getDisposition(),
+                                      std::make_unique<gui::ModesPopupData>(mode, flightMode));
+                }
+                return true;
+            });
+        popupBlueprint.registerBlueprint(
+            ID::Volume, [&](gui::popup::ID /*id*/, std::unique_ptr<gui::PopupRequestParams> &params) {
+                auto volumeParams = dynamic_cast<const gui::VolumePopupRequestParams *>(params.get());
+                if (volumeParams == nullptr) {
+                    return false;
+                }
+                LOG_INFO("Playback: %s, volume: %s",
+                         audio::str(volumeParams->getAudioContext().second).c_str(),
+                         std::to_string(volumeParams->getVolume()).c_str());
+                auto volume          = volumeParams->getVolume();
+                auto context         = volumeParams->getAudioContext();
+                const auto popupName = resolveWindowName(gui::popup::ID::Volume);
+                if (const auto currentWindowName = getCurrentWindow()->getName(); currentWindowName == popupName) {
+                    updateCurrentWindow(std::make_unique<gui::VolumePopupData>(volume, context));
+                }
+                else {
+                    switchWindowPopup(popupName,
+                                      volumeParams->getDisposition(),
+                                      std::make_unique<gui::VolumePopupData>(volume, context));
+                }
+                return true;
+            });
+        popupBlueprint.registerBlueprint(
+            ID::PhoneLock, [&](gui::popup::ID id, std::unique_ptr<gui::PopupRequestParams> & /*params*/) {
```

Comment:
I don't really like it that some of the windows specific functionalities are defined in some AppCommon counterpart. Please consider to split each of the parameters renderings to windows directories and register them only in here. Also "blueprint" name is kind of ambiguous here. Maybe `popupSwitchBlueprint` ?

Reply:
Tbh I do not want to refactor too much. It was taken out from ApplicationCommon - therefore its much easier to decouple from it further.
As for blueprint - we do not have any other blueprints here. I feel that we are tending to the point where we will names in style ThisIsVeryImportantClassWhichDoesSomething generally speaking such naming is bad practice. Tbh I would even name it `gui::popup::Blueprint` and function just `Register`